cpp/src/arrow/util/decimal_test.cc

```diff
@@ -985,6 +986,45 @@ TEST(Decimal128Test, Divide) {
   }
 }
 
+TEST(Decimal128Test, Rescale) {
+  ASSERT_EQ(Decimal128(11100), Decimal128(111).Rescale(0, 2).ValueOrDie());
+  ASSERT_EQ(Decimal128(111), Decimal128(11100).Rescale(2, 0).ValueOrDie());
+  ASSERT_EQ(Decimal128(5), Decimal128(500000).Rescale(6, 1).ValueOrDie());
+  ASSERT_EQ(Decimal128(500000), Decimal128(5).Rescale(1, 6).ValueOrDie());
+  ASSERT_TRUE(Decimal128(555555).Rescale(6, 1).status().IsInvalid());
+
+  // Test some random numbers.
+  Decimal128 multiplier(1);
+  for (int delta_scale = 0; delta_scale < 29;
+       delta_scale++, multiplier *= Decimal128(10)) {
```

Comment:
You can limit the cost of this test by making the `delta_scale` loop the inner loop (generating random numbers is slightly expensive).

Reply:
Changed accordingly. Thank you!